Assignment4.txt

```diff
@@ -0,0 +1,9 @@
+Clarke Littlejohn
+
+1) From what i understand there is not really a difference between the two but from readng online string points to a terminator,
+
+2)They can be useful for storing large amount of numbers or characters. There is no arrayIndexoutofbounds error message which can be annoying.
+
+3)No idea.
```

Comment:
you should always attempt questions! but the short answer is that a compiler doesn't generate the address whenever it appears in an expression.